### ISSUE TITLE
Change duration of multiple frames in animation editor

### ISSUE THREAD

**Issue by IMMZ** [bjorn/tiled#1310]
Sometimes in animation editor you want to change all frames' duration to some other value than default one. Doing it every frame when add them (new frames) or after adding could be annoying enough. It would be good to select some of or all the frames and change this value to what you need one. Also it could be useful to consider the next features:
- Add/Subtract[Multiply/Divide] selected durations by some number X.
- Set random duration values in range [X-Y] for selected frames.

Used ver: Version 2016.05.17

**Comment by Alturos** [comment]
Gonna have a crack at this. @bjorn: Thoughts on the bullet point items?

**Comment by bjorn** [comment]
@Alturos Before starting to implement it, please consider how the UI should work for this and make a proposal. As for the bullet points, I don't personally see these actions as particular common and hence not very useful. However, what I do think is frequently needed is to change the speed / duration of the entire animation, adjusting each frame's duration accordingly. Another thing to improve here is that Tiled currently always uses 100ms as the default frame duration, but I think this should be configurable.

Also note that on the `wip/tilesetdocument` branch, tileset editing has been moved away from map editing. This affects also the Tile Animation Editor. The code has already changed there, so modifying it on master will create conflicts, and actually I'd like to turn this animation editor dialog into a view of the new Tileset editing mode.

So hmm, after all this I'm not sure if this is the best moment to tackle this particular issue. :/

**Comment by Alturos** [comment]
Funny how that timing works. The proposal I'd come to (by getting it most of the way to done), was the simplest approach:

- Add a toolbar button to the row above the listview
- Enable/disable based on having a selection IN the listview
- Hitting the button pops up a small dialog with a spinnerbox
- - Entering a value and hitting OK updates all the selected entries.

More or less end of list. At this rate, it looks like I could have that done by the end of the weekend. It's pretty light as features go, if you want to slip it in ahead of the tileset refactor. If not, also cool, I'll table it and look for something else to poke at.

**Comment by bjorn** [comment]
Sounds fair. The other suggestions would of course be easy to add by means of a combo box in the dialog.

Actually it would be good to look at some other animation editors, to see how they work. I have personally never used an animation editor.

**Comment by Alturos** [comment]
I may have jumped the gun on this one, but i was already 3/4's done, so figured I should just finish it. I think it's in line with the general UX theme of Tiled in general (or so my weary 4AM mind tells me). But it was light enough work that I didn't mind doing it, even if it's just throw away. :smile: :sleepy:

**Comment by bjorn** [comment]
@satu0king Would you like to work on the other features suggested by @IMMZ or shall I move those to a new issue, so that we can close this one for claiming the bounty?

**Comment by satu0king** [comment]
I am not sure what bounty means. However, I think you can close the issue so as to avoid confusion. I am currently looking for issues that I can fix and get back to this eventually.

**Comment by bjorn** [comment]
@satu0king There is a $15 bounty on this issue at https://www.bountysource.com/issues/35278005-change-duration-of-multiple-frames-in-animation-editor, which you can now claim. :-)
